src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -256,6 +257,12 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
     //executorName is to distinguish different types of nodes
     //taskType is the type of task in mesos to launch on the node
     //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
```

Comment:
In the new code it exits the method without calling createExecutor if you don't actually need to create an executor. This may be related to why the executor didn't get terminated correctly.